tools/py/committee-reweight.py

```diff
@@ -0,0 +1,194 @@
+            obs_avg_CEA[j, i] = obs_avg[j] - obsxh_avg[j, i] + obs_avg[j] * h_avg[i]
+    return obs_avg_CEA, h_matrix
+
+
+def commitee_reweight(path2ixml, pot_file, obs_file, stride=1, index=-1, direct=False):
+    """
+    Parameters
+    ----------
+    pot_file    :   str, mandatory
+                    The file containing the values of the potential from each model, either in .dat form or as .extra.
+                    Each line contains the potentials for the M models for each frame
+    obs_file    :   str, mandatory
+                    A file containing the value of the observable for each frame.
+                    It is assumed that lines correspond to frames, while columns to properties.
+                    Multiple properties can be reweighted at the same time.
+    stride      :   integer, [1]
+                    The frequency of sampling of pot_file, if different from that of prop_file (e.g. --stride 10 if
+                    the potential was printed 10 times more often than the observable. --stride -10 must be used if
+                    the observable were printed more often than the potential).
+    index       :   integer, optional
+                    Which column of the property file should be used to compute the average and uncertainty.
+                    If it is not given, and there are multiple columns, they will be interpreted as corresponding
+                    to multiple property models
+    direct      :   bool, optional
+                    Activates the direct reweighting, instead of the cumulant expansion approximation.
+                    Use at your own risk!
+
+    """
+    if index >= 0:
+        obs = np.loadtxt(obs_file, usecols=index)
+    else:
+        obs = np.loadtxt(obs_file)
+
+
+    if stride > 0:
+        potentials = np.loadtxt(pot_file)[::stride]
+    elif stride < 0:
+        stride = np.abs(stride)
+        obs = obs[::stride]
+    else:
+        raise ValueError(
+                "Stride value cannot be zero"
+                )
+
+
+    # Load kbT from i-PI, we could make it into a small function
+    ifile = open(path2ixml, "r")
+    xmlrestart = io_xml.xml_parse_file(ifile)
+    ifile.close()
+
+    isimul = InputSimulation()
+    isimul.parse(xmlrestart.fields[0][1])
+    simul = isimul.fetch()
+
+    kbT = float(simul.syslist[0].ensemble.temp)
+    # CEA is the default choice. The weights or h_matrix are
+    if direct:
+        rw_obs, _weights = direct_reweight(potentials, obs, kbT)
+    else:
+        rw_obs, _h_matrix = CEA(potentials, obs, kbT)
+
+    np.savetxt("rw_" + obs_file, rw_obs)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="""This tool exploits a committee of ML models to compute the reweighted  canonical averages of a given observable, in order to quantify its uncertainty at thermodynamic equilibrium.
+    The units are assumed to be atomic, which are also the internal units of i-PI.
```

Comment:
hm! why not call the unit conversion functions of i-PI for more flexibility, etc.?

Reply:
this only really matters for pot, and we have no way of outputting committee pots as anything but atomic units. agree the text could be clearer

Reply:
Made clearer that the potential is expected to be in Hartree. This is meant as a warning for people who want to use this tool after running simulations with a different code